src/derivers/bip32.test.ts

```diff
@@ -2,16 +2,249 @@ import { bytesToHex, hexToBytes } from '@metamask/utils';
 import { CURVE } from '@noble/secp256k1';
 
 import fixtures from '../../test/fixtures';
-import { secp256k1 } from '../curves';
+import { BIP_32_HARDENED_OFFSET } from '../constants';
+import { ed25519, secp256k1 } from '../curves';
+import { SLIP10Node } from '../SLIP10Node';
 import { hexStringToBytes } from '../utils';
 import {
+  deriveChildKey,
   privateAdd,
   privateKeyToEthAddress,
   publicKeyToEthAddress,
 } from './bip32';
+import { bip39MnemonicToMultipath } from './bip39';
 
 const privateAddFixtures = fixtures['secp256k1-node'].privateAdd;
 
+describe('deriveChildKey', () => {
+  describe('private key derivation', () => {
+    it('handles invalid keys using BIP-32', async () => {
+      // TODO: Compare these results to reference implementations.
+
+      const node = await SLIP10Node.fromDerivationPath({
+        derivationPath: [bip39MnemonicToMultipath(fixtures.local.mnemonic)],
+        curve: 'secp256k1',
+        specification: 'bip32',
+      });
+
+      // Simulate an invalid key once.
+      jest.spyOn(secp256k1, 'isValidPrivateKey').mockReturnValueOnce(false);
+
+      const childNode = await deriveChildKey({
+        node,
+        path: `0'`,
+        curve: secp256k1,
+        specification: 'bip32',
+      });
+
+      expect(childNode.index).toBe(BIP_32_HARDENED_OFFSET + 1);
```

Comment:
Currently this returns the index for the next node. This needs to be compared to reference implementations, as they may return the index that was originally being derived, even though `n + 1` is being derived under the hood.